src/vs/editor/browser/controller/textAreaInput.ts

```diff
@@ -622,11 +640,12 @@ export class TextAreaInput extends Disposable {
 	}
 
 	public writeScreenReaderContent(reason: string): void {
-		if (this._currentComposition) {
+		if (((reason === 'asyncFocusGain' || reason === 'render') && !this._accessibilityService.isScreenReaderOptimized()) || this._currentComposition) {
```

Comment:
Should it return when simply `!this._accessibilityService.isScreenReaderOptimized()`?

Reply:
when I tried that originally, in non screen reader mode, things were broken when not in accessibility mode.